code/Controllers/SilverStripeNavigatorItem_ArchiveLink.php

```diff
@@ -46,10 +46,8 @@ public function getMessage()
 
     public function getLink()
     {
-        return Controller::join_links(
-            $this->record->PreviewLink(),
-            '?archiveDate=' . urlencode($this->record->LastEdited ?? '')
-        );
+        $link = $this->record->PreviewLink();
+        return $link ? Controller::join_links($link, '?archiveDate=' . urlencode($this->record->LastEdited ?? '')) : null;
```

Comment:
Probably, we need cover it with unit tests.
And also `join_links()` method returns empty string. I'm not sure, but if somebody uses this method and doesn't process NULL, then he probably would get an Exception.

Reply:
Good catch, I've used an empty string now instead. Unit tests also included,